crates/libs/windows/src/core/winsock.rs

```diff
@@ -0,0 +1,24 @@
+use windows::Win32::Networking::WinSock::*;
+
+/*
+#[cfg(feature = "Win32_Networking_WinSock")]
+impl core::convert::From<&std::net::Ipv4Addr> for IN_ADDR {
+    fn from(addr: &std::net::Ipv4Addr) -> IN_ADDR {
+        IN_ADDR { S_un: IN_ADDR_0 { S_addr: u32::from(addr) } }
+    }
+}
+*/
+
+#[cfg(feature = "Win32_Networking_WinSock")]
+impl core::convert::From<std::net::Ipv4Addr> for IN_ADDR {
```

Comment:
We could inject this code into the module that defines `IN_ADDR`. Not sure what the pros and cons are of the two approaches. If there aren't any for the consumer, then this is certainly simpler.

Reply:
Where is the module that defines `IN_ADDR`?  A cursory glance suggests `crates/libs/bindgen/src/extensions/` is what you're talking about?

I'm not a crate publishing expert, but I think this gets exported everywhere the same.  Simple is generally better, but what if the feature name changes?

Reply:
Yes, the main advantage is that it travels with the type. I'm now looking more seriously into splitting the `windows` crate into smaller chunks and that would allow such extensions to automatically get generated into the appropriate crate.

Reply:
I think that's a very worthwhile goal. I'm having to do all this in vim in a terminal because my laptop (only 16 GB RAM) can't get rust-analyzer to finish parsing the project.